frontends/web/tests/helpers/regtest.ts

```diff
@@ -0,0 +1,180 @@
+/**
+*  Copyright 2025 Shift Crypto AG
+*
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*       http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*/
+
+import { exec, spawn, ChildProcess } from 'child_process';
+import path from 'path';
+import util from 'util';
+import fs from 'fs';
+
+
+const execAsync = util.promisify(exec);
+
+const RPC_USER = 'dbb';
+const RPC_PASSWORD = 'dbb';
+const RPC_PORT = 10332;
+const DATADIR = '/bitcoin';
+
+let addr: string;
+
+// run bitcoin-cli command inside the bitcoind-regtest docker container.
+async function runBitcoinCli(args: string[]): Promise<string> {
+  const cmd = [
+    'docker exec',
+    '--user=$(id -u)',
+    'bitcoind-regtest',
+    'bitcoin-cli',
+    '-regtest',
+    `-datadir=${DATADIR}`,
+    `-rpcuser=${RPC_USER}`,
+    `-rpcpassword=${RPC_PASSWORD}`,
+    `-rpcport=${RPC_PORT}`,
+    ...args
+  ].join(' ');
+
+  const { stdout } = await execAsync(cmd);
+  return stdout.trim();
+}
+
+// Setup regtest by
+// - Creating a wallet
+// - Getting a new address
+// - Generating 101 blocks to that address
+export async function setupRegtestWallet(): Promise<void> {
+  await runBitcoinCli(['createwallet', 'testwallet']);
+  addr = await runBitcoinCli(['getnewaddress']);
+  await runBitcoinCli(['generatetoaddress', '101', addr]);
+}
+
+// mineBlocks mines the given number of blocks to the regtest wallet address.
+// This is useful to confirm transactions in tests.
+export async function mineBlocks(numBlocks: number): Promise<void> {
+  await runBitcoinCli(['generatetoaddress', numBlocks.toString(), addr]);
+}
+
+/**
+ * Send coins to a given address in the regtest wallet.
+ * @param address The address to send to
+ * @param amount The amount in BTC
+ */
+export async function sendCoins(address: string, amount: number | string): Promise<string> {
+  // bitcoin-cli expects amount as a string with decimal point
+  const amt = typeof amount === 'number' ? amount.toFixed(8) : amount;
+  const txid = await runBitcoinCli(['sendtoaddress', address, amt]);
+  return txid; // returns the transaction ID
+}
+
+
+export function launchRegtest(): Promise<ChildProcess> {
+
+  // First, clean up cache and headers.
+  try {
+    const basePath = path.resolve(__dirname, '../../../../appfolder.dev/cache');
```

Comment:
Would it help to use nodejs built in `process.cwd()` function here instead of traversing up the directory structure?

https://nodejs.org/docs/latest/api/process.html#processcwd